src/app/common/psbt/use-psbt-request-params.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import * as btc from '@scure/btc-signer';
+
+import { useDefaultRequestParams } from '../hooks/use-default-request-search-params';
+
+export function usePsbtRequestSearchParams() {
+  const [searchParams] = useSearchParams();
+  const { origin, tabId } = useDefaultRequestParams();
+  const requestToken = searchParams.get('request');
+  return useMemo(
+    () => ({
+      origin,
+      tabId: tabId ?? 1,
+      requestToken,
+    }),
+    [origin, requestToken, tabId]
+  );
+}
+
+export function useRpcSignPsbtParams() {
+  const [searchParams] = useSearchParams();
+  const { origin, tabId } = useDefaultRequestParams();
+  const requestId = searchParams.get('requestId');
+  const psbtHex = searchParams.get('hex');
+  const publicKey = searchParams.get('publicKey');
+  const allowedSighash = searchParams.getAll('allowedSighash');
+  const signAtIndex = searchParams.getAll('signAtIndex');
```

Comment:
Not for now, but wondering if we should just encode JSON into a single param, than doing these separately. Avoids the complexities between `get/getAll` etc